src/utils/format.ts

```diff
@@ -132,6 +133,38 @@ function _decomposeBn(amount: BN, amountPrecision: number, decimals: number): {
   return { integerPart, decimalPart, decimalsPadded }
 }
 
+/**
+ * Transforms (possibly decimal) amount as string into a BN
+ * Takes into account original precision from amount + argument precision
+ *
+ *
+ * Internally we use BNs, but we all know BNs don't like decimals.
+ * To convert arbitrary strings, we need to make them integers.
+ * We do so by creating a BigNumber and moving the decimal separator to the right
+ * by multiplying it by 10^(amount precision).
+ * Which we then later add to regular precision to be 'compressed'.
+ * Finally, convert it to a string base 10,
+ * to avoid numbers with exponents which BN also doesn't like
+ *
+ * @param amountStr Amount with arbitrary precision
+ * @param amountPrecision precision on top of existing precision
+ */
+function _stringToBn(amountStr:string, amountPrecision:number):{ amount: BN, precision: number } {
```

Comment:
Makes sense. Will expose it.

As for the parameter names, the reason for it is that `amount` and `precision` are exactly what I'm returning.
So I changed to avoid confusion between input/output.